iothub/device/src/InternalClient.cs

```diff
@@ -57,58 +50,46 @@ internal class InternalClient : IDisposable
 
         private int _diagnosticSamplingPercentage;
 
-        protected internal InternalClient(
+        internal InternalClient(
             IotHubConnectionCredentials iotHubConnectionCredentials,
-            IotHubClientOptions iotHubClientOptions,
-            IDeviceClientPipelineBuilder pipelineBuilder)
+            IotHubClientOptions iotHubClientOptions)
         {
-            Argument.AssertNotNull(iotHubClientOptions, nameof(iotHubClientOptions));
-
             if (Logging.IsEnabled)
-                Logging.Enter(this, iotHubClientOptions.TransportSettings, pipelineBuilder, nameof(InternalClient) + "_ctor");
+                Logging.Enter(this, iotHubClientOptions?.TransportSettings, nameof(InternalClient) + "_ctor");
+
+            // Make sure client options is initialized.
+            if (iotHubClientOptions == default)
+            {
+                iotHubClientOptions = new();
+            }
 
             IotHubConnectionCredentials = iotHubConnectionCredentials;
             _clientOptions = iotHubClientOptions;
 
-            if (!iotHubClientOptions.ModelId.IsNullOrWhiteSpace()
-                && iotHubClientOptions.TransportSettings is IotHubClientHttpSettings)
+            if (!_clientOptions.ModelId.IsNullOrWhiteSpace()
+                && _clientOptions.TransportSettings is IotHubClientHttpSettings)
             {
                 throw new InvalidOperationException("Plug and Play is not supported over the HTTP transport.");
             }
 
-            var pipelineContext = new PipelineContext
+            _pipelineContext = new PipelineContext
             {
```

Comment:
The internal client now sets up the common properties in the pipeline context. The device client and module client will set up their specific properties and then build the pipeline.